main.go

```diff
@@ -159,10 +169,18 @@ func main() {
 		WorkerRouterClient:    workerRouterClient,
 		WorkerNetOpClient:     workerNetOPClient,
 		EventRecorder:         sliceGwEventRecorder,
+		NodeIP:                nodeIP,
```

Comment:
I don't think we need to pass the dynamic NodeIP from there. The same will be fetched by `r.reconcileNodes` method. Instead, we should have a separate variable for NodeIP from env variable.

something like:

```
ProvidedNodeIP: os.Env("NODE_IP")
```

in slicegatewayreconciler, it should check if this variable is set (which means Node IP is provided as env variable) and then ignore the reconciliation. You don't even have to start NodeReconciler if this is set.

Reply:
but what if the provided nodeIP is no longer available? do we not want to select another nodeIP in that case?